src/lib/utils/styling/styler.ts

```diff
@@ -0,0 +1,128 @@
+ */
+import {Inject, Injectable, Optional, PLATFORM_ID} from '@angular/core';
+import {isPlatformBrowser, isPlatformServer} from '@angular/common';
+
+import {applyCssPrefixes} from '../auto-prefixer';
+import {ServerStylesheet} from './server-stylesheet';
+import {SERVER_TOKEN} from './server-token';
+
+@Injectable()
+export class StyleService {
+
+  constructor(@Optional() private _serverStylesheet: ServerStylesheet,
+              @Optional() @Inject(SERVER_TOKEN) private _serverModuleLoaded: boolean,
+              @Inject(PLATFORM_ID) private _platformId) {}
+
+  /**
+   * Applies styles given via string pair or object map to the directive element
+   */
+  applyStyleToElement(element: HTMLElement, style: StyleDefinition, value?: string | number) {
+    let styles = {};
+    if (typeof style === 'string') {
+      styles[style] = value;
+      style = styles;
+    }
+    styles = applyCssPrefixes(style);
+    this._applyMultiValueStyleToElement(styles, element);
+  }
+
+  /**
+   * Applies styles given via string pair or object map to the directive's element
+   */
+  applyStyleToElements(style: StyleDefinition, elements: HTMLElement[] = []) {
+    const styles = applyCssPrefixes(style);
+    elements.forEach(el => {
+      this._applyMultiValueStyleToElement(styles, el);
+    });
+  }
+
+  /**
+   * Determine the DOM element's Flexbox flow (flex-direction)
+   *
+   * Check inline style first then check computed (stylesheet) style
+   */
+  getFlowDirection(target: HTMLElement): [string, string] {
+    const query = 'flex-direction';
+    let value = this.lookupStyle(target, query);
+    if (value === FALLBACK_STYLE) {
+      value = '';
+    }
+    const hasInlineValue = this.lookupInlineStyle(target, query) ||
+    (isPlatformServer(this._platformId) && this._serverModuleLoaded) ? value : '';
+
+    return [value || 'row', hasInlineValue];
+  }
+
+  /**
+   * Find the DOM element's raw attribute value (if any)
+   */
+  lookupAttributeValue(element: HTMLElement, attribute: string): string {
+    return element.getAttribute(attribute) || '';
+  }
+
+  /**
+   * Find the DOM element's inline style value (if any)
+   */
+  lookupInlineStyle(element: HTMLElement, styleName: string): string {
+    return element.style[styleName] || element.style.getPropertyValue(styleName);
+  }
+
+  /**
+   * Determine the inline or inherited CSS style
+   * @TODO(CaerusKaru): platform-server has no implementation for getComputedStyle
+   */
+  lookupStyle(element: HTMLElement, styleName: string, inlineOnly = false): string {
+    let value = '';
+    if (element) {
+      let immediateValue = value = this.lookupInlineStyle(element, styleName);
+      if (!immediateValue) {
+        if (isPlatformBrowser(this._platformId)) {
+          if (!inlineOnly) {
+            value = getComputedStyle(element).getPropertyValue(styleName);
+          }
+        } else {
+          if (this._serverModuleLoaded) {
```

Comment:
There are a lot of places throughout the PR where you're explicitly performing a check for browser vs. server and performing a different action for each. In general it would be better to delegate to some injectable with a common interface that uses a different implementation on each platform.

Reply:
Now that the `MatchMedia` has been changed (see above), the only three places this still happens are the following: ImgSrc, ShowHide, and Styler. In each case, the check only changes about two lines of code in very critical ways. I think it would duplicate a lot of code to change and as it stands now doesn't benefit much to factor out.